### PR TITLE
feat: support vue sfc

### LINKED ISSUE #18: [Question] Vue/Nuxt Support
I tried to make this plugin work with Nuxt, but failed so far. Is that something what can be configured or is it not supported?

### PR DESCRIPTION
**Pull request by DanSnow**
# Pull Request Name

## Description

This PR add supporting for Vue SFC file

Fixes #18 

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce.
Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

**Test Configuration**:
* OS + version: e.g. macOS Mojave
* NPM version: ...
* Node version: ...

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have checked my code and corrected any misspellings

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -1,8 +1,3029 @@
 {
   "name": "eslint-plugin-tailwindcss",
   "version": "1.10.0",
-  "lockfileVersion": 1,
+  "lockfileVersion": 2,
```
Comment (francoismassart) [inline package-lock.json:4]: Looks like you are using a newer version of node.

As of today the LTS version of node is:

> Latest LTS Version: 14.17.0 (includes npm 6.14.13)

While you might be using:

> Latest Current Version: 16.3.0 (includes npm 7.15.1)

About the [`lockfileVersion`](https://docs.npmjs.com/cli/v7/configuring-npm/package-lock-json#lockfileversion)

- `1`: The lockfile version used by `npm v5` and `v6`.
- `2`: The lockfile version used by `npm v7`, which is backwards compatible to `v1` lockfiles.

I think we should stick to `v1` for now.

**Review by francoismassart — CHANGES_REQUESTED** [review]

File: lib/util/vue.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
+const path = require('path');
+
+function defineTemplateBodyVisitor(context, templateBodyVisitor, scriptVisitor) {
+  if (context.parserServices == null || context.parserServices.defineTemplateBodyVisitor == null) {
```
[the remark this answers is not in the document]
Comment (DanSnow) [reply lib/util/vue.js:4]: `context.parserService` is an extra property added by  `vue-eslint-parser`. I discovered this by reading the source code of `eslint-plugin-vue`

File: lib/util/vue.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+    const filename = context.getFilename();
+    if (path.extname(filename) === '.vue') {
+      context.report({
```
[the remark this answers is not in the document]
Comment (DanSnow) [reply lib/util/vue.js:7]: Looks like syntax error will emit first. I don't think that warning will able to show now.

File: lib/util/vue.js [hunk 4]
```diff
@@ -0,0 +1,20 @@
+const path = require('path');
```
[the remark this answers is not in the document]
Comment (DanSnow) [reply lib/util/vue.js:1]: Will do this

File: package-lock.json (hunk 1 above)
Comment (DanSnow) [reply package-lock.json:4]: Will do it.

File: lib/util/vue.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (DanSnow) [reply lib/util/vue.js:4]: Just noticed that `parserServices` is also documented at [here](https://eslint.org/docs/developer-guide/working-with-rules#the-context-object)

**Comment by DanSnow** [comment]
@francoismassart Thanks for reviewing. I have fixed above issues.

**Review by francoismassart — APPROVED** [review]

**Pull request merged**